luigi/execution_summary.py

```diff
@@ -0,0 +1,377 @@
+
+
+def _get_str_one_parameter(tasks):
+    row = ''
+    count = 0
+    for task in tasks:
+        if (len(row) >= 30 and count > 2 and count != len(tasks) - 1) or len(row) > 200:
+            row += '...'
+            break
+        row += '{0}'.format(getattr(task, task.get_params()[0][0]))
+        if count < len(tasks) - 1:
+            row += ','
+        count += 1
+    return row
+
+
+def _serialize_first_param(task):
+    return task.get_params()[0][1].serialize(getattr(task, task.get_params()[0][0]))
+
+
+def _get_number_of_tasks(task_dict):
+    num = 0
+    for task_family, tasks in task_dict.items():
+        num += len(tasks)
+    return num
+
+
+def _get_comments(group_tasks):
+    comments = {}
+    for status, task_dict in group_tasks.items():
+        comments[status] = "* {0}".format(_get_number_of_tasks(task_dict))
+        if _get_number_of_tasks(task_dict) == 0:
+            comments.pop(status)
+    if "already_done" in comments:
+        comments["already_done"] += ' present dependencies were encountered:\n'
+    if "completed" in comments:
+        comments["completed"] += ' ran successfully:\n'
+    if "failed" in comments:
+        comments["failed"] += ' failed:\n'
+    still_pending = False
+    if "still_pending_ext" in comments:
+        comments["still_pending_ext"] = '    {0} were missing external dependencies:\n'.format(comments['still_pending_ext'])
+        still_pending = True
+    if "upstream_run_by_other_worker" in comments:
+        comments["upstream_run_by_other_worker"] = '    {0} had dependencies that were being run by other worker:\n'.format(comments['upstream_run_by_other_worker'])
+        still_pending = True
+    if "upstream_failure" in comments:
+        comments["upstream_failure"] = '    {0} had failed dependencies:\n'.format(comments['upstream_failure'])
+        still_pending = True
+    if "upstream_missing_dependency" in comments:
+        comments["upstream_missing_dependency"] = '    {0} had missing external dependencies:\n'.format(comments['upstream_missing_dependency'])
+        still_pending = True
+    if "run_by_other_worker" in comments:
+        comments["run_by_other_worker"] = '    {0} were being run by another worker:\n'.format(comments['run_by_other_worker'])
+        still_pending = True
+    if "unknown_reason" in comments:
+        comments["unknown_reason"] = '    {0} were left pending because of unknown reason:\n'.format(comments["unknown_reason"])
+        still_pending = True
+    if still_pending:
+        comments["still_pending"] = '* {0} were left pending, among these:\n'.format(_get_number_of_tasks(group_tasks["still_pending_ext"]) + _get_number_of_tasks(group_tasks["still_pending_not_ext"]))
+    return comments
+
+
+def _get_statuses():
+    statuses = ["already_done", "completed", "failed", "still_pending", "still_pending_ext", "run_by_other_worker", "upstream_failure", "upstream_missing_dependency", "upstream_run_by_other_worker", "unknown_reason"]
```

Comment:
The statuses are repeated in other parts of the code (_partition_tasks). Should only be hardcoded in one place. Using a class would have been a whole lot easier.